src/hvm.cu

```diff
@@ -126,9 +126,15 @@ struct RBag {
   Pair lo_buf[RLEN];
 };
 
+#include "shared_mem_config.h"
+
+#ifndef HVM_SHARED_MEM
+#define HVM_SHARED_MEM 0x2000 // Default value
+#endif
+
 // Local Net
-const u32 L_NODE_LEN = 0x2000;
-const u32 L_VARS_LEN = 0x2000;
+const u32 L_NODE_LEN = HVM_SHARED_MEM;
```

Comment:
Aren't there places in the code that expect L_NODE_LEN to have this exact value?

If you change it I think we'll start to get either memory leaks, out of bounds access or non local memory use.

Reply:
Also, if this number is smaller then default, the programs compiled by bend won't run.

If it's smaller the performance will also tank incredibly fast. (see the issue where someone halved this number and got worse performamce than in the cpu)

Reply:
> Aren't there places in the code that expect L_NODE_LEN to have this exact value?

i wouldn't say that it expects to have this value, but rather that it expects to have the numbers of a device with 8.9 of compute capability, but then again, this is not a general optimization of the CUDA version, this is just to make the allocation of shared mem on local net dynamic, so that users can install the hvm without having to manually change it on the `hvm.cu`, the rest of the code is still the same, made with the numbers of a 4090 in mind. We can slowly improve the code to make it `device` based, instead of hard-coded to the specs of a `4090`.

> Also, if this number is smaller then default, the programs compiled by bend won't run.

can you show me an example of what was ran and what was the device/numbers used when this happened?


> If it's smaller the performance will also tank incredibly fast. (see the issue where someone halved this number and got worse performamce than in the cpu)

i mean, that's of course, if you give it less memory, it will have less memory, besides the fact that the rest of the code is 'optimized' for a `4090`.

we can like, whenever someone installs it using a GPU with <99KB of max shared mem per block, we give a warning saying something along the lines of: 
"HVM is currently optimized to run on devices with >=96KB of max shared mem, please be aware that your GPU performance will be reduced dramatically"

Reply:
> can you show me an example of what was ran and what was the device/numbers used when this happened?



> > Also, if this number is smaller then default, the programs compiled by bend won't run.
> 
> can you show me an example of what was ran and what was the device/numbers used when this happened?

The number of nodes in the local buffer determines the maximum size of hvm definitions. If we decrease `L_NODE_LEN` then Bend also needs to generate programs with smaller maximum definition size.